### DISCUSSION TITLE
How can I edit the CMakeLists.txt in my project with the Dht library added?

### DISCUSSION THREAD

**Discussion by se-ka1500** [UncleRus/esp-idf-lib#267] [Q&A] [answered]
I am using Esp32 devkit V1 in my project. I couldn't find how to revise the CmakeLists.txt files in esp-idf. The error I get when I build idf.py is as follows:

```

/home/.../workspace/esp_projects/test_dht11/main/main.c:4:10: fatal error: dht.h: No such file or directory
 #include <dht.h>
          ^~~~~~~
compilation terminated.
make[2]: *** [esp-idf/main/CMakeFiles/__idf_main.dir/build.make:76: esp-idf/main/CMakeFiles/__idf_main.dir/main.c.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:4670: esp-idf/main/CMakeFiles/__idf_main.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
make failed with exit code 2
```
I used dht.c and dht.h files in the same way in the dht folder. I just set the necessary places for me to use dht11.
My project folder is as below : 
```
- main
      main.c 
      CMakeLists.txt
- core
      -inc
               dht.h 
               esp_idf_lib_helpers.h

      -src
               dht.c
      -CMakeLists.txt
-build
-CMakeLists.txt
       
```

My top level CMakeList.txt  is as follows : 
```
cmake_minimum_required(VERSION 3.5)

set(EXTRA_COMPONENT_DIRS $ENV{IDF_PATH}/home/../../../test_dht11/core)

include($ENV{IDF_PATH}/tools/cmake/project.cmake)
project(test_dht11)


```
Inside my core file CMakeList.txt is like this.I've intentionally commented some places here. Honestly, it shouldn't be like this. I left it that way as I couldn't figure out how to edit it. This part confused me the most.
```
#if(${IDF_TARGET} STREQUAL esp8266)
##    set(req esp8266 freertos log esp_idf_lib_helpers)
##else()
##    set(req driver freertos log esp_idf_lib_helpers)
##endif()
set(sources 
    src/dht.c

)

idf_component_register(
    SRCS ${sources}
    INCLUDE_DIRS "inc"
    REQUIRES driver
            freertos
)

```

Inside my main file CMakeList.txt is like this 
```
idf_component_register(SRCS "main.c"
                    INCLUDE_DIRS ".")

```

Thanks in advance for your help.

**Comment by trombik** [comment]
i cannot solve your issue because:

* you omitted logs
* you did not follow the instruction in `README.md` in this repository
* some explanations do not make sense (you have three `CMakefile.txt`, which one are you referring to?)

it seems that you are not very familiar with `cmake`, or compiling C in general. if so, start from the example provided in `esp-idf`. the official documentation also recommends to copy an example, and modify it one by one. [tree](http://mama.indstate.edu/users/ice/tree/) is a useful tool that you should have in your toolbox.

**Comment by UncleRus** [comment, marked as answer]
Hi!
You don't need to copy files from esp-idf-lib to your project. You can add whole esp-idf-lib as submodule.
See for example this project: https://github.com/UncleRus/EvLamp
